packages/headless/src/features/question-answering/question-answering-slice.ts

```diff
@@ -82,6 +82,9 @@ export const questionAnsweringReducer = createReducer(
         state.feedbackModalOpen = false;
       })
       .addCase(executeSearch.fulfilled, (state, action) => {
+        if (!action.payload.response.questionAnswer) {
```

Comment:
Possible that since it goes through a different endpoint (search-interface as opposed to directly SearchAPI) that's where the difference comes from ?

If it's true that `questionAnswer` is possibly undefined when coming back from the API, a better fix would be to change `search-response.ts` to mark it as possibly undefined ( `questionAnswer?:  QuestionAnswers`), and fix all the compilation problem that will arise from the type change.

Reply:
I confirmed it is indeed possible that `questionAnswer` is undefined in a searchapi response. It's the case when no smart snippets model is associated to the pipeline used. I'll try your suggestion and see what those compilation issues are. Just want to make sure we aren't doing anything breaking.

Reply:
If we previously marked that part of the state as always being defined and we make it optional that could be breaking no?

Reply:
What do you think?

Reply:
With the new changes, this could be removed, right ?